plugin/code_actions.py

```diff
@@ -71,6 +73,9 @@ def request_factory(session: Session) -> Optional[Request]:
             return Request.codeAction(params, view)
 
         def response_filter(session: Session, actions: List[CodeActionOrCommand]) -> List[CodeActionOrCommand]:
+            self.refactor_actions_cache.extend(
+                [(session.config.name, cast(CodeAction, a)) for a in actions
+                    if not is_command(a) and kinds_include_kind([CodeActionKind.Refactor], a.get('kind'))])
```

Comment:
It's not fully correct to cache and get refactors from the standard region request because standard request doesn't include "only: [refactor]" and servers can and do return more refactors when "refactor" kind is explicitly specified. `typescript-language-server` does that for example.

Triggering "refactor..." from the context menu includes it but request that happens on moving the cursor doesn't.

Reply:
Hm, that's unfortunate. I guess we don't want to send another request with `"only": ["refactor"]` everytime when the cursor/selection changes. So if other servers also don't send the "refactor" actions in the response to the regular request after selection change, then this PR is probably useless for those servers. On the other hand, the implementation here is a strict addition to the current behavior, because currently there isn't any such item in the "Edit" main menu at all.

It's probably worth to check more servers, if they include the "refactor" actions or not.

Reply:
> It's probably worth to check more servers, if they include the "refactor" actions or not.

To be fair, I'm not sure if I came up with this logic for `typescript-language-server` or if that's something that I've read about or some elsewhere... Maybe it's not exactly expected to return different refactor results based on `only`...